### DISCUSSION TITLE
Backlog Quota have to be smaller than Retention

### DISCUSSION THREAD

**Discussion by mnit016** [apache/pulsar#21086] [Q&A] [answered]
Hi everyone, I had a discussion for above topic before [#discussion_17697](https://github.com/apache/pulsar/discussions/17697), but today I have more question about this one:

I'm thinking when messages are on retention, when we create a new subscription with Earliest position, it will bring all messages from retention to Backlog right? Whereas backlog have to be smaller than retention size? 
Example:
1. create new topic with backlog 10KB, retention 1MB
2. Send larger than 10KB of messages -> (go directly to retention)
3. Create new Subscription with Earliest position -> it will bring all messages back to backlog and cause full backlog -> block all current producers.
![image](https://github.com/apache/pulsar/assets/37934995/711376b9-5132-4f85-8e37-99c679f5574a)

Could someone please explain: Why backlog shouldn't be larger than retention for above scenario? And what's exact block if backlog larger than rentention?

Hi @Jason918 Sorry if any inconvenience, I'd like to ping you directly here, you said "if backlog quota is larger than retention, it won't take effect" in previous discussion.
Did you mean "**retention won't take effect** or **backlog quota won't take effect**"?

Thanks much

**Comment by Jason918** [comment, marked as answer]
> Did you mean "retention won't take effect or backlog quota won't take effect"?

It's `backlog quota won't take effect`.
Because even if you set a very large backlog quota, the messages over retention policy would be deleted anyway, so the backlog quota won't take effect.

**Reply by mnit016** [reply]
oh got it, basically it's logic inside Pulsar. Thanks @Jason918
